### DISCUSSION TITLE
GSw_CoalRetire switch in Cases.csv file

### DISCUSSION THREAD

**Discussion by ssattler** [NREL/ReEDS-2.0#60] [Q&A: Features]
Can you tell me more about the GSw_CoalRetire switch in the Cases.csv file? It says that it "adjusts the lifetime of coal retirements", how does it do this? Does it make changes to endogenous retirements of coal, or take a different a different lifetime scenario?
Thanks.

**Comment by wesleyjcole** [comment]
The switch is used in b_inputs to set the retirement dates for coal generators.  Setting the switch to 1 decreases coal plant lifetimes by 10 years if that plant was going to retire in 2030 or later.  Setting the switch to 2 extends the coal lifetimes by 10 years if the plant was going to retire in 2022 or later.

Here is the code from b_inputs:

`*If coal retirement switch is 1, shorten coal lifetimes by 10 years if the plant retires in 2030 or later`
`if(Sw_CoalRetire = 1,`
`     prescribedretirements(allt,r,i,"value")$[coal(i)$(allt.val >= 2030)] = 0 ;`
`     prescribedretirements(allt,r,i,"value")$coal(i) = prescribedretirements_copy(allt+10,r,i,"value")$(allt.val >= 2030-10) + prescribedretirements(allt,r,i,"value") ;`
`) ;`

`*If coal retirement switch is 2, extend coal lifetimes by 10 years if the plant retires in 2022 or later`
`if(Sw_CoalRetire = 2,`
`     prescribedretirements(allt,r,i,"value")$[coal(i)$(allt.val >= 2022)] = 0 ;`
`     prescribedretirements(allt,r,i,"value")$[coal(i)$(allt.val >= 2022)] = prescribedretirements_copy(allt-10,r,i,"value")$(allt.val >= 2022+10) ;`
`) ;`

**Reply by ssattler** [reply]
Thanks - could you tell me more about the different settings for the GSw_Retire switch?

**Reply by wesleyjcole** [reply]
These switch settings are described in the readme:

![image](https://user-images.githubusercontent.com/49044852/134193613-ea600feb-1d1e-4e7f-a86a-8b7e1c7900f6.png)

You can also search the code for Sw_Retire, where you'll find it used in d_solveprep.

**Reply by ssattler** [reply]
Thank you!

**Reply by wanyingw193** [reply]
Hi, may I know the meaning of GSw_RetireYear? For example, the code related to GSw_RetireYear in d_solveprep is as follows. Then if the default input is 2022, how will the model correctly represent the retirements from 2010 to 2021? Thanks!
-----code below-----------------
*Do not allow retirements before they are allowed
retiretech(i,v,r,t)$[(yeart(t)<retireyear)] = no ;

**Reply by wesleyjcole** [reply]
GSw_RetireYear is the first year that _endogenous_ retirements are allowed.  So if the value is set to 2022, then all retirements happening before 2022 will be based on official retirement dates and will not occur due to model decision-making.

**Reply by wanyingw193** [reply]
thank you! it is clear!
